libsurfman/src/surface.c

```diff
@@ -78,6 +78,26 @@ static void surface_update_mfn_list (surfman_surface_t * surface)
   free(pfns);
 }
 
+static void surface_update_mfn_arr (surfman_surface_t * surface)
+{
+  struct surface_priv *p = PRIV(surface);
+  xen_pfn_t *pfns;
+  int rc;
+  int domid = surface->pages_domid;
+  size_t i, n = surface->page_count;
+
+  assert(p->type == TYPE_PFN_ARR);
+
+  pfns = xcalloc(n, sizeof (xen_pfn_t));
+  for (i = 0; i < n; ++i)
+    pfns[i] = p->u.pfn_arr.arr[i];
```

Comment:
`memcpy`?

Reply:
I agree, the other surface manipulating surfaces did not do so, so I carried over the loop.

Would a new commit amending everything at once (following this change set) be preferable?